samples/.NET/NUnit/PageTestWithArtifact.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Azure;
+using Microsoft.Playwright;
+using Microsoft.Playwright.NUnit;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+
+namespace PlaywrightTests
+{
+    /*
+     * PageTestWithArtifact is a base class for tests that require Playwright page and context.
+     * It provides a setup and teardown methods for the test to run in a browser context.
+     * It also provides a way to enable trace, screenshot, and video artifacts for the test.
+     * Users need to inherit this class instead of PageTest to write tests.
+     */
+    [TestFixture]
+    public class PageTestWithArtifact : PageTest
+    {
+
+        [SetUp]
+        public async Task Setup()
+        {
+            TestContext.WriteLine("Browser: " + BrowserName);
+            // Enable Trace
+            await Context.Tracing.StartAsync(new()
+            {
+                Title = $"{TestContext.CurrentContext.Test.Name}",
```

Comment:
This code breaks sometimes as we have seen if test case name is more than 255 characters then in windows since file name is restricted to a certain length... so we might have to introduce a check to trim

Reply:
sure, make sense, i will look for existing pattern and update it.

Reply:
I thought about it and added comment, dont want to make the logic complex as its boilerplate code.